nntrainer/src/tensor.cpp

```diff
@@ -775,4 +811,58 @@ Tensor Tensor::standardization() const {
 
   return result;
 }
+
+ExternalLoopInfo Tensor::computeExternalLoop(const Tensor &m) {
+  if (m.length() > this->length())
+    throw exception::not_supported("broadcasting *this is not supported");
+
+  const TensorDim m_dim = m.getDim();
+
+  ExternalLoopInfo e;
+
+  for (unsigned int i = 0; i < MAXDIM; ++i) {
+    if (m_dim.getTensorDim(i) == 1) {
+      e.strides[i] = 0;
+      continue;
+    }
+
+    if (dim.getTensorDim(i) == m_dim.getTensorDim(i)) {
+      e.strides[i] = m.strides[i];
+      continue;
+    }
+
+    std::stringstream ss;
+    ss << "[computeExternalLoop] given shape is not broadcastable,\n"
+       << "this: " << dim << "target: " << m_dim;
+    throw std::invalid_argument(ss.str().c_str());
+  }
+
+  /// calculate inner loop size
+  unsigned int inner_loop_size = dim.getTensorDim(3);
+
+  bool first_dim_none = m_dim.getTensorDim(3) == 1;
+  e.buffer_axis = -1;
+  for (int axis = 2; axis >= 0; --axis) {
+    unsigned int cur_dim = m_dim.getTensorDim(axis);
+
+    if (first_dim_none) {
+      if (cur_dim != 1) {
+        e.buffer_axis = axis;
+        break;
+      }
+    } else {
+      if (cur_dim == 1) {
```

Comment:
there should also be a check here to match `dim.getTensorDim(axis) == m_dim.getTensorDim(axis)`??
Consider the case of dim = (3,5,1,4) and m_dim = (3,1,1,4)
e.buffer_axis = 2 and I think that's what not is intended.

Reply:
Nice catch. 
I added test case for dim = (3, 5, 1, 4) and m_dim = (1, 5, 1, 4) on which buffer size should be 20.